src/pow.rs

```diff
@@ -8,21 +8,59 @@ use std::{
 };
 
 use arrayvec::ArrayVec;
+use bdk_wallet::bitcoin::Amount;
 use concurrent_map::{CasFailure, ConcurrentMap};
 use parking_lot::{Mutex, MutexGuard};
 use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
 use sha2::{Digest, Sha256};
 use terrors::OneOf;
 use tokio::time::sleep;
 
-use crate::err;
+use crate::{err, settings::SETTINGS};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Challenge {
     nonce: Nonce,
+    claimed: bool,
     expires_at: Instant,
+    difficulty: u8,
 }
 
-const TTL: Duration = Duration::from_secs(60);
+#[derive(Deserialize, Serialize, Debug)]
+pub struct PowConfig {
+    /// Minimum balance required for a user to claim funds. Defaults to 500 BTC,
+    /// or 50_000_000_000 sats. When configuring in the config file, this value
+    /// should be in sats as a number.
+    pub min_balance: Amount,
+    /// Minimum difficulty required for a user to claim funds. Defaults to 17.
+    ///
+    /// Users will have to solve a POW challenge witha chance of finding of
+    /// 1 / 2^min_difficulty per random guess. The faucet will dynamically adjust
+    /// the actual difficulty given to the user based on the current balance,
+    /// min_balance and sats_per_claim.
+    pub min_difficulty: u8,
+    /// How long a challenge is valid for. Defaults to 60 seconds.
+    ///
+    /// In config, this should be provided as an object with fields `secs` and `nanos`.
+    /// For example:
+    ///
+    /// ```toml
+    /// [pow]
+    /// challenge_duration = { secs = 60, nanos = 0 }
+    /// ```
```

Comment:
```suggestion
    /// How long a challenge is valid for.
    ///
    /// Defaults to `60` seconds.
    ///
    /// In config, this should be provided as an object with fields `secs` and `nanos` with integers.
    /// For example:
    ///
    /// ```toml
    /// [pow]
    /// challenge_duration = { secs = 60, nanos = 0 }
    /// ```
```